lib/puppet/parser/scope.rb

```diff
@@ -638,11 +638,9 @@ def handle_not_found(class_name, variable_name, position, reason = nil)
     unless Puppet[:strict_variables]
       # Do not issue warning if strict variables are on, as an error will be raised by variable_not_found
       location = if position[:lineproc]
-                   " at #{position[:lineproc].call}"
-                 elsif position[:file] && position[:line]
-                   " at #{position[:file]}:#{position[:line]}"
+                   Puppet::Util::Errors.error_location_with_space("", position[:lineproc].call)
                  else
-                   ""
+                   Puppet::Util::Errors.error_location_with_space(position[:file], position[:line])
```

Comment:
If file and line are not set, then the method will return an empty string so we don't need the else case.